### ISSUE TITLE
Two strange questions in  keras_siamese_example.py in Keras

### ISSUE THREAD

**Issue by ztf-ucas** [keras-team/keras#10532]
Hi, folks. I am trying running keras_siamese_example.py in Keras, but I met two strange questions: 
1. Why there are two different loss?
I just added a callback class to see the output log, but the loss and accuracy in output log are different from the default output?
My code is written as follows:
<pre><code>
from keras.callbacks import Callback
class myhistory(Callback):
    def on_batch_end(self, batch, logs=None):
        # print('my loss: ', logs.get('loss'))
        print('logs: ', logs)
</code></pre>

<pre><code>
model.fit([tr_pairs[:, 0], tr_pairs[:, 1]], tr_y,
          batch_size=128,
          epochs=epochs,
          validation_data=([te_pairs[:, 0], te_pairs[:, 1]], te_y),
          callbacks=[myhistory()])
</code></pre>

Please refer to [https://github.com/keras-team/keras/blob/master/examples/mnist_siamese.py](url) for full code.
My output aer as follows:
<pre><code>
2944/108400 [..............................] - ETA: 39s - loss: 0.2803 - accuracy: 0.5686logs:  {'loss': 0.22223043, 'accuracy': 0.6796875, 'size': 128, 'batch': 22}

  3072/108400 [..............................] - ETA: 40s - loss: 0.2782 - accuracy: 0.5706logs:  {'loss': 0.23038688, 'accuracy': 0.6171875, 'size': 128, 'batch': 23}

  3200/108400 [..............................] - ETA: 40s - loss: 0.2766 - accuracy: 0.5722logs:  {'loss': 0.2366159, 'accuracy': 0.609375, 'size': 128, 'batch': 24}
logs:  {'loss': 0.23526536, 'accuracy': 0.59375, 'size': 128, 'batch': 25}

  3456/108400 [..............................] - ETA: 39s - loss: 0.2727 - accuracy: 0.5770logs:  {'loss': 0.21242097, 'accuracy': 0.6796875, 'size': 128, 'batch': 26}
logs:  {'loss': 0.20234957, 'accuracy': 0.640625, 'size': 128, 'batch': 27}
logs:  {'loss': 0.22053646, 'accuracy': 0.6796875, 'size': 128, 'batch': 28}
</code></pre>
I try to set batch_size=1, but it doesn't work.
<pre><code>
55/108400 [..............................] - ETA: 53:20 - loss: 0.4956 - accuracy: 0.4727logs:  {'size': 1, 'accuracy': 0.0, 'loss': 0.89840335, 'batch': 54}
logs:  {'size': 1, 'accuracy': 1.0, 'loss': 0.037929423, 'batch': 55}

    57/108400 [..............................] - ETA: 53:29 - loss: 0.4835 - accuracy: 0.4737logs:  {'size': 1, 'accuracy': 0.0, 'loss': 0.26763284, 'batch': 56}
logs:  {'size': 1, 'accuracy': 0.0, 'loss': 0.43662825, 'batch': 57}

    59/108400 [..............................] - ETA: 54:03 - loss: 0.4851 - accuracy: 0.4576logs:  {'size': 1, 'accuracy': 0.0, 'loss': 0.62083471, 'batch': 58}
logs:  {'size': 1, 'accuracy': 1.0, 'loss': 0.0021861708, 'batch': 59}

    61/108400 [..............................] - ETA: 54:28 - loss: 0.4804 - accuracy: 0.4590logs:  {'size': 1, 'accuracy': 0.0, 'loss': 0.68298244, 'batch': 60}
logs:  {'size': 1, 'accuracy': 1.0, 'loss': 0.10688201, 'batch': 61}

    63/108400 [..............................] - ETA: 54:47 - loss: 0.4686 - accuracy: 0.4762logs:  {'size': 1, 'accuracy': 1.0, 'loss': 0.11206865, 'batch': 62}
logs:  {'size': 1, 'accuracy': 0.0, 'loss': 0.56470454, 'batch': 63}
</code></pre>

2.Why the loss value dropped sharply, when from the first epoch to the second epoch?
<pre><code>
107264/108400 [============================>.] - ETA: 0s - loss: 0.0955 - accuracy: 0.8867
107904/108400 [============================>.] - ETA: 0s - loss: 0.0953 - accuracy: 0.8871
108400/108400 [==============================] - 17s 158us/step - loss: 0.0951 - accuracy: 0.8873 - val_loss: 0.0472 - val_accuracy: 0.9501
Epoch 2/2

   128/108400 [..............................] - ETA: 13s - loss: 0.0424 - accuracy: 0.9609
   768/108400 [..............................] - ETA: 10s - loss: 0.0501 - accuracy: 0.9531
  1408/108400 [..............................] - ETA: 9s - loss: 0.0478 - accuracy: 0.9581 
  2048/108400 [..............................] - ETA: 9s - loss: 0.0487 - accuracy: 0.9556
</code></pre>

**Comment by brge17** [comment]
@1, 2: 
The loss displayed is an average of the loss from each batch.  E.g. if the loss of batch 1 is 10, batch 2 is 5, and batch 3 is 3 the loss will display 6 after the third batch. 

The loss is significantly higher during the beginning of the first epoch which brings the average up.

**Comment by ztf-ucas** [comment]
@brge17 
Thank you very much！I understand！So I just need to pay attention to the loss in batch_logs.
